plutus-contract/src/Language/Plutus/Contract/StateMachine.hs

```diff
@@ -198,8 +201,9 @@ waitForUpdateUntil StateMachineClient{scInstance, scChooser} timeoutSlot = do
                 then go (succ sl)
                 else pure txns
 
+    initial <- currentSlot
+    txns <- go initial
```

Comment:
Why this change?

Reply:
It was a bug. I used this `initial slot` to compare with `timeout` slot in `case states of`, and it could not ever `Timeout`.

Reply:
Sounds like it deserves a comment!